test/node-test/client-dispatch.js

```diff
@@ -938,8 +939,9 @@ test('dispatches in expected order for http2', async (t) => {
       onBodySent () {
         dispatches.push('onBodySent')
       },
-      onResponseStarted () {
-        dispatches.push('onResponseStarted')
```

Comment:
Can you add a test that keeps showing this behavior? Or add a new test for onStart?